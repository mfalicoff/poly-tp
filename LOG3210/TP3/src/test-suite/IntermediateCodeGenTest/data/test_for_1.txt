Declaration
num a;
num i;
EndDeclaration

a = 0;

for (i = 0; i < 3; i = i + 1) {
  a = 4 * i;
}
