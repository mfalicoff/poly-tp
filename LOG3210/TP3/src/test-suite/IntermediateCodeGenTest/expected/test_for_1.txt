a = 0
_L1
i = 0
_L2
if i < 3 goto _L4
goto _L0
_L4
_t0 = 4 * i
a = _t0
_L3
_t1 = i + 1
i = _t1
goto _L2
_L0
